### PR TITLE
Rewrite non-libuv tcp server and drop libuv

### PR DESCRIPTION
**Pull request by thestr4ng3r**
# DO NOT SQUASH

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Despite being a fantastic library, libuv was only ever used for the tcp server for almost 4 years. This fixes the non-libuv tcp server and drops libuv in favor of it. See commit descriptions for details.

**Test plan**

Warning: if `tcp.islocal=false`, the following will open complete shell access to `0.0.0.0`!
`rz -c "Rt 1337"`, then from another terminal do e.g. `echo "?E hello" | nc localhost 1337` and see the command output.
Ctrl+C should also break the `Rt` command properly.

### CODE REVIEW

File: librz/core/rtr.c [hunk 1]
```diff
@@ -1023,251 +998,81 @@ RZ_API bool rz_core_rtr_init(RZ_NONNULL RzCore *core) {
...
 	}
-
-	RzSocket *s = rz_socket_new(0);
 	s->local = rz_config_get_i(core->config, "tcp.islocal");
 
 	if (!rz_socket_listen(s, port, NULL)) {
-		eprintf("Error listening on port %s\n", port);
-		rz_socket_free(s);
-		return false;
+		RZ_LOG_ERROR("Error listening on port %s\n", port);
```
Comment (wargio) [inline librz/core/rtr.c:1013]: INFO?

Comment (wargio) [inline librz/core/rtr.c]: Cannot listen on port XXXX

File: librz/socket/socket.c [hunk 2]
```diff
@@ -663,7 +670,8 @@ RZ_API bool rz_socket_block_time(RzSocket *s, bool block, int sec, int usec) {
 		return false;
 	}
 #elif __WINDOWS__
-	ioctlsocket(s->fd, FIONBIO, (u_long FAR *)&block);
+	u_long winblock = !block; // 0 = blocking, 1 = non-blocking on windows
```
Comment (wargio) [inline librz/socket/socket.c]: don't assume !0 = 1, do `block ? 0 : 1;`

File: librz/core/rtr.c [hunk 3]
```diff
@@ -1023,251 +998,81 @@ RZ_API bool rz_core_rtr_init(RZ_NONNULL RzCore *core) {
...
-	if (status < 0) {
-		eprintf("New connection error: %s\n", uv_strerror(status));
-		return;
-	}
-
-	rtr_cmds_context *context = server->loop->data;
-
-	uv_tcp_t *client = RZ_NEW(uv_tcp_t);
-	if (!client) {
+RZ_API void rz_core_rtr_cmds(RzCore *core, const char *port) {
```
Comment (XVilka) [inline librz/core/rtr.c:1001]: Doxygen, please

File: librz/socket/socket.c [hunk 4]
```diff
@@ -716,7 +724,7 @@ RZ_API char *rz_socket_to_string(RzSocket *s) {
 	}
 	return str;
 #else
-	return NULL;
+		return NULL;
```
Comment (XVilka) [inline librz/socket/socket.c:727]: Is this indentation intended?

File: librz/socket/socket.c [hunk 5]
```diff
@@ -898,4 +912,160 @@ RZ_API ut8 *rz_socket_slurp(RzSocket *s, int *len) {
...
 
+struct rz_stop_pipe_t {
+#if __WINDOWS__
+	WSAEVENT event;
+#else
+	int fds[2];
+#endif
+}; // RzStopPipe
+
+RZ_API RzStopPipe *rz_stop_pipe_new(void) {
```
Comment (XVilka) [inline librz/socket/socket.c:923]: Doxygen with explanation what exactly RzStopPipe is?

**Review by XVilka — CHANGES_REQUESTED** [review]
Resend this from `dist-asan-fuzz-` branch, please.
@GustavoLCR could you please take a look if everything is fine from the Windows point of view?

File: librz/socket/socket.c (hunk 4 above)
Comment (thestr4ng3r) [reply librz/socket/socket.c:727]: clang-format enforces it, but it makes no sense imo.

**Comment by thestr4ng3r** [comment]
Superseded by #2860

**Pull request closed without merging**